service/controller/resource/deployment/deployment.go

```diff
@@ -77,3 +88,39 @@ func getVPNSubnet(customObject providerv1alpha1.AzureConfig) (*net.IPNet, error)
 
 	return &subnets.VPN, nil
 }
+
+func (r *Resource) initialProvisioning(ctx context.Context, customObject providerv1alpha1.AzureConfig) (string, error) {
+	subnetsClient, err := r.clientFactory.GetSubnetsClient(customObject.Spec.Azure.CredentialSecret.Namespace, customObject.Spec.Azure.CredentialSecret.Name)
+	if err != nil {
+		return IsInitialProvisioning, microerror.Mask(err)
+	}
+
+	result, err := subnetsClient.ListComplete(ctx, key.ClusterID(&customObject), key.VnetName(customObject))
+	if err != nil {
+		return IsInitialProvisioning, microerror.Mask(err)
+	}
+
+	expectedSubnets := 0
+	for result.NotDone() {
+		subnet := result.Value()
+		if strings.Contains(*subnet.Name, key.VNetGatewaySubnetName()) {
+			expectedSubnets++
+		} else if strings.Contains(*subnet.Name, "MasterSubnet") {
+			expectedSubnets++
+		} else if strings.Contains(*subnet.Name, "WorkerSubnet") {
+			expectedSubnets++
+		}
+
+		err = result.NextWithContext(ctx)
+		if err != nil {
+			return IsInitialProvisioning, microerror.Mask(err)
+		}
+	}
+
+	initialProvisioning := "No"
+	if expectedSubnets != 3 {
```

Comment:
What if it is 4 or more because there are 4 or more node pools?

Reply:
I'm only increasing if it matches the expected names. If we create more subnets for nodepools for example, they won't use the same name, hence they won't increase `expectedSubnets`.

Reply:
Oh yes, missed that.